### ISSUE TITLE
Update Ruby, Rails

### ISSUE THREAD

**Issue by amykirschner** [oscurrency/oscurrency#625]
- [ ] @ziyan-junaideen can you start making a list of what needs to be done. Thank you!

**Comment by lesliev** [comment]
Hi people! Has anyone actually gotten oscurrency running on Rails 4? I started with the `upgrade/rails4` branch tonight - fixed a bunch of things but still not completely running. I hope I'm not duplicating anyone's work.

**Comment by lesliev** [comment]
I have a strange error here:
`From: /home/leslie/dev/git/other/external/oscurrency4/db/seeds.rb @ line 95 :`
`state = State.find_or_create_by_name( :name => value[0], :abbreviation => 'AL' )`

I hardcoded 'AL' there because it seemed the abbreviation was not being set.

Error:
```
[1] pry(#<ActiveRecord::ConnectionAdapters::PostgreSQLAdapter>)> e
=> #<PG::NotNullViolation: ERROR:  null value in column "abbreviation" violates not-null constraint
DETAIL:  Failing row contains (1, Alabama, null, 2019-04-10 10:18:09.577495, 2019-04-10 10:18:09.577495).
```

SQL:
`INSERT INTO "states" ("created_at", "name", "updated_at") VALUES ($1, $2, $3) RETURNING "id"`

As you can see, the abbreviation column is not getting to the SQL. Making me think "abbreviation" might be a reserved name that gets stripped out or something.

I've moved to pg v 0.21.0 and Ruby 2.3.7, so that might be something to do with it.

Anyway I'll work some more later in the week, let me know if anyone has any ideas.

**Comment by amykirschner** [comment]
Hi @lesliev, 

We were headed for Rails 5 about a year ago but we haven't moved forward much since then. I believe my older version is still on 3. I'm not a developer, I'm a project mgr on the business side so sorry I can't answer that in detail. 

Here's where we were in 2018: http://oscurrency.org/2018/02/21/community-raising-crowdfunding-code-for-cooperation/

Maybe can answer your question: @herestomwiththeweather - @ziyan-junaideen @devinbalkind

**Comment by lesliev** [comment]
Yes, it's on 3. A little work seems to have been done to get it on 4. I'd like to get it on 5 too!

**Comment by herestomwiththeweather** [comment]
cool! fwiw, i just tried a dev install with upgrade/rails4 branch for the first time (after commenting out a bunch attr_accessible's and attr_protected's and routes) and i didn't have a problem with seeding the states using ruby 2.1.5 and pg 0.15.1 (arbitrarily chosen) - i ran rake install after rake db:create.  i can't think of what might be causing the error.

**Comment by lesliev** [comment]
Thanks, good to know. Probably Ruby then. I've not been able to compile such an old Ruby on my Ubuntu 18.04 because of an SSL issue. I've been reluctant to try and get that compiling because I probably want a currently supported Ruby anyway.

**Comment by lesliev** [comment]
The problem was caused by me adding `gem 'protected_attributes'` to try and bypass having to strong-params all the controllers. But models such as State and FeePlan didn't have any `attr_accessible`s defined. I want to get tests passing and the app running as soon as possible so I have a working base to start incrementally fixing. So I *think* the shortest path forwards is to add missing `attr_accessible`'s, get tests to pass, *then* to switch to strong params and try to convert the attrs to `permit` calls.

**Comment by lesliev** [comment]
I've made some good progress tonight, tests are actually running. They all fail because of some problem with fixtures, but I'm getting there. When I'm done, how should I commit this? Should I fork and commit to my own repo?

**Comment by herestomwiththeweather** [comment]
that seems like a good plan as it would allow a pull request.

**Comment by lesliev** [comment]
Ok. I'm making steady progress. The install task is running and I can start the server and log in. There are a fair number of things to upgrade for Rails 4, I'm working through ActiveRecord API changes. 
Authlogic-Openid seems a bit broken, I'll come back to it.

**Comment by herestomwiththeweather** [comment]
cool! i think there's a preference to disable openid.
